### PR TITLE
tests(turbo): Rust CLI Tests

### PR DESCRIPTION
**Pull request by NicholasLYang**
Adding tests for both CLI parsing and e2e.

### CODE REVIEW

File: shim/src/main.rs [hunk 1]
```diff
@@ -390,13 +440,214 @@ mod test {
...
+            expected_output: Args {
+                command: Some(Command::Prune {
+                    scope: Some("foo".to_string()),
+                    docker: true,
+                    output_dir: "dist".to_string(),
+                }),
+                ..Args::default()
+            },
+        }
+        .test();
```
Comment (mehulkar) [inline shim/src/main.rs:651]: nit: is it convention to put all rust tests in the same file (as opposed to `.test.rs` or something? Feels pretty noisy/distracting to have this here

Comment (mehulkar) [reply shim/src/main.rs:651]: I see `shim/tests/mod.rs`. What's the difference between those tests and these?

**Review by mehulkar — APPROVED** [review]
Question about Rust but otherwise looks like an improvement!

File: shim/src/main.rs (hunk 1 above)
Comment (NicholasLYang) [reply shim/src/main.rs:651]: It's a difference of integration versus unit tests. These are tests that require internal data structures like `Args` that we wouldn't have access to in `tests/`. I agree that this file is starting to get unwieldy though. Might split it up and put the repo state code in its own file.

**Comment by mehulkar** [comment]
Also, do we need a GH workflow to run these tests or does that already exist somewhere?

File: shim/tests/mod.rs [hunk 2]
```diff
@@ -1,5 +1,69 @@
 use assert_cmd::Command;
 
+static TURBO_HELP: &str = "turbo
```
Comment (chris-olszewski) [inline shim/tests/mod.rs:3]: Commented on #2386, but I generally prefer to keep UI snapshot tests in something like cram/prysk that has better support for updating snapshots support for diffing snapshots.

File: shim/src/main.rs [hunk 3]
```diff
@@ -338,6 +348,46 @@ fn main() -> Result<()> {
...
+
+    impl CommandTestCase {
+        fn test(&self) {
+            let permutations = self.create_all_arg_permutations();
+            for command in permutations {
+                assert_eq!(Args::try_parse_from(command).unwrap(), self.expected_output)
+            }
+        }
+
+        fn create_all_arg_permutations(&self) -> Vec<Vec<&'static str>> {
```
Comment (chris-olszewski) [inline shim/src/main.rs:368]: Should we also be testing interspersing global args with the command args? If I'm reading this correctly we're only putting global args right after `turbo`. IIRC correctly cobra allows for these args to appear anywhere which is a behavior we want to keep.

**Review by chris-olszewski — APPROVED** [review]
Biggest concern is that we're not testing commands that have global args in a position other than right after `turbo`

**Pull request merged**